assignment7.txt

```diff
@@ -0,0 +1,25 @@
+1.
+The difference between ++*P, *p++ and *++p is that:
+++*P increases the value of *p by one
+*p++ moves up the location of the pointer by one
+*++p will do exactly what ++*p does
+
+2. 
+Yes, the order is guaranteed
+
+3.
+For fuction parameters
+for dynamic programming
+Pointer enables you to do things that are otherwise impossible, it also speeds up the operation
+
+4.
+4.1 invalid because b and c are undeclared.
+4.2 invalid because of undeclared variables.
```

Comment:
This is actually valid! "xyz" is just an array of characters, so the "xyz"[1] is just accessing "y". Then you just subtract 'y' from it to get 0. This is because, if you can recall, a char is just a value mapped to a character!